Sources/SpeziFirebaseAccount/FirebaseAccountService.swift

```diff
@@ -0,0 +1,890 @@
+    private var signInWithApple = FirebaseSignInWithAppleButton()
+
+    /// Security alert to authorize security sensitive operations.
+    ///
+    /// This view modifier injects an alert into the view hierarchy that will present an alert, if the account service requests to re-authenticate the
+    /// user for security-sensitive operations. This modifier is automatically injected in SpeziAccount-related views.
+    @SecurityRelatedModifier public var securityAlert = FirebaseSecurityAlert()
+
+    @Model private var firebaseModel = FirebaseAccountModel()
+    @Modifier private var firebaseModifier = FirebaseAccountModifier()
+
+    @MainActor private var authStateDidChangeListenerHandle: AuthStateDidChangeListenerHandle?
+    @MainActor private var lastNonce: String?
+
+    private var shouldQueue = false
+    private var queuedUpdates: [UserUpdate] = []
+    private var actionSemaphore = AsyncSemaphore()
+    private var skipNextStateChange = false
+
+
+    private var unsupportedKeys: AccountKeyCollection {
+        var unsupportedKeys = account.configuration.keys
+        unsupportedKeys.removeAll(Self.supportedAccountKeys)
+        return unsupportedKeys
+    }
+
+    /// - Parameters:
+    ///   - providers: The authentication methods that should be supported.
+    ///   - emulatorSettings: The emulator settings. The default value is `nil`, connecting the FirebaseAccount module to the Firebase Auth cloud instance.
+    ///   - passwordValidation: Override the default password validation rule. By default firebase enforces a minimum length of 6 characters.
+    public init(
+        providers: FirebaseAuthProviders,
+        emulatorSettings: (host: String, port: Int)? = nil,
+        passwordValidation: [ValidationRule]? = nil // swiftlint:disable:this discouraged_optional_collection
+    ) {
+        self.emulatorSettings = emulatorSettings
+
+        self.configuration = AccountServiceConfiguration(supportedKeys: .exactly(Self.supportedAccountKeys)) {
+            RequiredAccountKeys {
+                \.userId
+                if providers.contains(.emailAndPassword) {
+                    \.password
+                }
+            }
+
+            UserIdConfiguration.emailAddress
+            FieldValidationRules(for: \.userId, rules: .minimalEmail)
+            FieldValidationRules(for: \.password, rules: passwordValidation ?? [.minimumFirebasePassword])
+        }
+
+        if !providers.contains(.emailAndPassword) {
+            $loginWithPassword.isEnabled = false
+        }
+        if !providers.contains(.signInWithApple) {
+            $signInWithApple.isEnabled = false
+        }
+        if providers.contains(.anonymousButton) {
+            $anonymousSignup.isEnabled = true
+        }
+    }
+
+    @_documentation(visibility: internal)
+    public func configure() {
+        if let emulatorSettings {
+            Auth.auth().useEmulator(withHost: emulatorSettings.host, port: emulatorSettings.port)
+        }
```

Comment:
I could think of a FirebaseEmulator configuration Module. E.g., for simple configurations one could reuse the same host and just supply the port for all the different types of emulators. However, we should preserve maximum flexibility to configure each emulator with a different host if you want to.

I will create an issue for that. Might make sense to create a PR before we tag the final 2.0 release so this can be a breaking change.